compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -2917,91 +2933,534 @@ setRootConfigImpl(mlir::FunctionOpInterface entryPointFn, Operation *op,
+            rootLoweringConfig.getTilingLevelAttr(level));
+        sizes.assign(attr.getSizes().begin(), attr.getSizes().end());
+        // Only `VectorCommonParallel` has scalable flags.
+        flags.assign(attr.getScalableFlags().begin(),
+                     attr.getScalableFlags().end());
+      }
+    } else {
+      if (rootLoweringConfig.hasTilingLevel(level)) {
+        sizes =
+            rootLoweringConfig.getStaticTilingLevelSizes(level, rootOperation);
+        flags.resize(sizes.size(), false);
+      }
+    }
+
+    // `MultiLoweringConfigGenerator` propagates tiling on the unpacked
+    // dimensions, while `rootLoweringConfig` defines tiling on the packed
+    // inner dimensions. Therefore, use inverse
+    // `scaleAndPermutateTilingForPackOp` to translate tiling information from
+    // the packed back to the unpacked dimensions.
+    if (auto packOp = dyn_cast<linalg::PackOp>(rootOperation);
+        packOp && !sizes.empty()) {
+      scaleAndPermutateTilingForPackOp(packOp, sizes, flags,
+                                       /*inverse=*/true);
+    }
+
+    // Map the tiling information from the op-level local dimension indices
+    // to dispatch-region global dimension indices.
+    globalTileSizes[level].assign(totalLoopNum, 0);
+    globalScalableTileFlags[level].assign(totalLoopNum, false);
+    for (auto [pos, size, flag] : llvm::enumerate(sizes, flags)) {
+      int64_t globalDimIdx = dimTracker.getGlobalDimIdx(rootOperation, pos);
+      globalTileSizes[level][globalDimIdx] = size;
+      globalScalableTileFlags[level][globalDimIdx] = flag;
+    }
+  };
+
+  // Load all tiling levels.
+  for (int i = 0, e = static_cast<int>(TilingLevel::MaxNumTileLevels); i < e;
+       ++i) {
+    loadTilingLevel(static_cast<TilingLevel>(i));
   }
-  // Pack op requires special vector tile sizes to achieve good performance.
-  // Override the parallel vector tile sizes from pack op.
-  auto vecTileSizes = getPackVectorTileSizes(entryPointFn, packOp);
-  auto outerDimsPerm = packOp.getOuterDimsPerm();
-  if (!outerDimsPerm.empty()) {
-    auto invertedPerm = invertPermutationVector(outerDimsPerm);
-    applyPermutationToVector(vecTileSizes, invertedPerm);
+}
+
+void MultiLoweringConfigGenerator::getVecTileSizesForNonRootOps(
+    mlir::FunctionOpInterface entryPointFn) {
+  for (auto op : computeOps) {
+    // Tile sizes have been initialized from the root op, so skip it.
+    if (op == rootOperation) {
+      continue;
+    }
+    if (auto packOp = dyn_cast<linalg::PackOp>(op)) {
+      nonRootOpVecTileSizes[op] =
+          getVecTileSizesForNonRootPackOp(entryPointFn, packOp);
+    } else if (auto unpackOp = dyn_cast<linalg::UnPackOp>(op)) {
+      nonRootOpVecTileSizes[op] = getVecTileSizesForNonRootUnPackOp(unpackOp);
+    } else if (auto genericOp = dyn_cast<linalg::GenericOp>(op)) {
+      nonRootOpVecTileSizes[op] =
+          getVecTileSizesForNonRootGenericOp(entryPointFn, genericOp);
+    }
```

Comment:
Do we add an assertion in `else` branch? Or at least we should log what ops are ignored in the processing. I think having a log looks better. Sometimes CPU backend works without "ideal" lowering config propagation, and it is valuable.